hyperparameter_tuning/tensorflow2_mnist/hpo_tensorflow2_mnist.ipynb

```diff
@@ -0,0 +1,370 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "objective_metric_name = 'average test loss'\n",
+    "objective_type='Minimize'\n",
+    "metric_definitions = [\n",
+    "    {\n",
+    "        'Name': 'average test loss',\n",
+    "        # checkout line 161 in code/train.py\n",
+    "        'Regex': 'Test Loss: ([0-9\\\\.]+)' \n",
+    "    }\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we'll create a `HyperparameterTuner` object, to which we pass:\n",
+    "- The `TensorFlow` estimator we created above\n",
+    "- Our hyperparameter ranges\n",
+    "- Objective metric name and definition\n",
+    "- Tuning resource configurations such as Number of training jobs to run in total and how many training jobs can be run in parallel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tuner = HyperparameterTuner(\n",
+    "    est,\n",
+    "    objective_metric_name,\n",
+    "    hyperparamter_range,\n",
+    "    metric_definitions,\n",
+    "    max_jobs=4,\n",
+    "    max_parallel_jobs=2,\n",
+    "    objective_type=objective_type\n",
+    ")\n",
+    "\n",
+    "tuner.fit(inputs=channels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy the Best Model\n",
+    "After training with hyperparameter optimization we can deploy the best performing model (by the objective metric we defined above) to a SageMaker endpoint. For more information about deploying a model to a SageMaker endpoint, checkout [this example on deploying a TensorFlow model](https://sagemaker-examples.readthedocs.io/en/latest/frameworks/tensorflow/get_started_mnist_deploy.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = tuner.deploy(\n",
+    "    initial_instance_count=1,\n",
+    "    instance_type='ml.m5.xlarge'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Evaluate\n",
+    "Now, we can evaluate the best performing model by invoking the endpoint with MNIST test set. The test data needs to be readily consumable by the model. So we need to arrange them into the correct shape that can be accepted by a TensorFlow model. Moreoever, we need to normalize it so that the pixel values have mean 0 and standard deviation 1, since this is the convention we used to train the model."
```

Comment:
```suggestion
    "Now, you can evaluate the best performing model by invoking the endpoint with MNIST test set. The test data needs to be readily consumable by the model. So you need to arrange them into the correct shape that can be accepted by a TensorFlow model. Moreover, you need to normalize it so that the pixel values have mean 0 and standard deviation 1, since this is the convention you used to train the model."
```